src/devices/RelayBoard/Relay.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.RelayBoard
+{
+    /// <summary>
+    /// A generic GPIO relay device.
+    /// </summary>
+    public class Relay : IDisposable
+    {
+        private readonly GpioController _controller;
+
+        #region Properties
+
+        private bool _shouldDispose;
+
+        /// <summary>
+        /// The pin this relay uses.
+        /// </summary>
+        public int Pin { get; }
+
+        /// <summary>
+        /// Gets the type of relay this device is.
+        /// </summary>
+        public RelayType Type { get; }
+
+        /// <summary>
+        /// Get or set if the relay is currently on
+        /// </summary>
+        public bool On
+        {
+            get
+            {
+                // TODO: make sure this isn't backwards.
+                return Type == RelayType.NormallyClosed ? _controller.Read(Pin) == PinValue.High : _controller.Read(Pin) == PinValue.Low;
+            }
+            set
+            {
+                // TODO: make sure this isn't backwards.
+                var val = Type == RelayType.NormallyClosed ? value : !value;
+                _controller.Write(Pin, val);
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Relay"/> class.
+        /// </summary>
+        /// <param name="pin">The pin this relay is on.</param>
+        /// <param name="gpioController">The GPIO controller to use.</param>
+        /// <param name="relayType">The type of relay this is.</param>
+        /// <param name="pinNumberingScheme">Numbering scheme to use for <paramref name="gpioController"/>, if one is not passed.</param>
+        /// <param name="shouldDispose">Whether this class should dispose the <paramref name="gpioController"/>.</param>
+        public Relay(int pin, GpioController? gpioController = null, RelayType relayType = RelayType.NormallyClosed, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = gpioController == null || shouldDispose;
+            Pin = pin;
+            Type = relayType;
+            _controller = gpioController ?? new GpioController(pinNumberingScheme);
+            _controller.OpenPin(pin);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            _controller?.ClosePin(Pin);
+
+            if (_shouldDispose)
+            {
+                _controller?.Dispose();
+            }
```

Comment:
Looking at some other implementations in the Iot.Device.* namespace, it seems like the convention is to null out the controller instance after disposing it. Also, it looks like GpioController [closes any open pins on dispose](https://github.com/dotnet/iot/blob/master/src/System.Device.Gpio/System/Device/Gpio/GpioController.cs#L343), so manually closing before disposal isn't necessary

I suggest this implementation from [DhtBase](https://github.com/dotnet/iot/blob/06fc3a0e78f74e430e26183ad342b24c1c92c368/src/devices/Dhtxx/DhtBase.cs#L287)
```suggestion
            if (_shouldDispose)
            {
                _controller?.Dispose();
                _controller = null;
            }
            else if(_controller?.IsPinOpen(Pin) ?? false)
            {
                _controller.ClosePin(Pin); 
            }
```